File: Source/Zybach.API/Controllers/ReportTemplateNewDto.cs
using System.ComponentModel.DataAnnotations;
using Microsoft.AspNetCore.Http;


namespace Zybach.API.Controllers
{
    public class ReportTemplateNewDto
    {
        public int ReportTemplateID { get; set; }
        [Required]
        public IFormFile FileResource { get; set; }
        [Required]
        public string DisplayName { get; set; }
        public string Description { get; set; }
        [Required]
        public int ReportTemplateModelID { get; set; }
    }
}

File: Source/Zybach.Web/src/app/shared/models/report-template-new-dto.ts
import { FileResourceDto } from "./generated/file-resource-dto"
import { ReportTemplateModelDto } from "./generated/report-template-model-dto"
import { ReportTemplateModelTypeDto } from "./generated/report-template-model-type-dto"

export class ReportTemplateNewDto {
	ReportTemplateID : number
	DisplayName : string
	Description : string
	ReportTemplateModelID : number
    FileResource: File
}